src/main/sensors/barometer.c

```diff
@@ -315,54 +314,6 @@ void baroSetGroundLevel(void)
     baroSetCalibrationCycles(SET_GROUND_LEVEL_BARO_CYCLES);
 }
 
-#define PRESSURE_SAMPLES_MEDIAN 3
-
-static int32_t applyBarometerMedianFilter(int32_t newPressureReading)
-{
-    static int32_t barometerFilterSamples[PRESSURE_SAMPLES_MEDIAN];
-    static int currentFilterSampleIndex = 0;
-    static bool medianFilterReady = false;
-    int nextSampleIndex;
-
-    nextSampleIndex = (currentFilterSampleIndex + 1);
-    if (nextSampleIndex == PRESSURE_SAMPLES_MEDIAN) {
-        nextSampleIndex = 0;
-        medianFilterReady = true;
-    }
-
-    barometerFilterSamples[currentFilterSampleIndex] = newPressureReading;
-    currentFilterSampleIndex = nextSampleIndex;
-
-    if (medianFilterReady)
-        return quickMedianFilter3(barometerFilterSamples);
-    else
-        return newPressureReading;
-}
-
-static uint32_t recalculateBarometerTotal(uint32_t pressureTotal, int32_t newPressureReading)
-{
-    static int32_t barometerSamples[BARO_SAMPLE_COUNT_MAX + 1];
-    static int currentSampleIndex = 0;
-    int nextSampleIndex;
-
-    // store current pressure in barometerSamples
-    if (currentSampleIndex >= barometerConfig()->baro_sample_count) {
-        nextSampleIndex = 0;
-        baroReady = true;
-    } else {
-        nextSampleIndex = (currentSampleIndex + 1);
-    }
-    barometerSamples[currentSampleIndex] = applyBarometerMedianFilter(newPressureReading);
```

Comment:
median filter was removing baro outliers. Removing it may cause considerable problems for some configuration

Reply:
Yes, though the median filter implementation appears to cause a lot of problems as well.

Reply:
Problem is that it is difficult to test new code without median. The problem may appear only with some kind of baro ...
And 3 point median adds typically 1 sample delay. Doesn't sound to bad

Reply:
The main difficulty in the old code was that the raw Pressure reading could not be effectively logged.  BBE would not show altitude raw from Baro, only pressure raw. 
Pressure raw is a number like 35,000 with about integer steps of 20-30 for 1m of elevation.
BBE would scale this in a way that made the graphs useless.
I don't quite know how it was originally tested.
To directly compare behaviour would require a special branch of master where we convert the raw and smoothed Pressure readings to altitudes so that the logs can be directly compared.
Should I make such a branch?

Reply:
They were using some kind of 21 point median.  And so far I don't see spikes, just noise.  
I have tested by blowing hard and fast on the baro, which does generate spikes.  
The images above show how it is handled.

Reply:
- I don't think that comparison to old code is necessary.
- there are lot of barometers and FC supported. Some may generate spikes (or maybe there was bug in readout in past and median masked this). I would keep 3-point median at this point
- 21point boxcar is overkill, IIR filter shall be beter here (and more tunable)